### PR TITLE
Initialised Channels Discussions and added crud operations for it

### LINKED ISSUE #23: make controllers for channels

### LINKED ISSUE #24: make controllers for Channel Discussions(messages) + announcements +questions

### PR DESCRIPTION
**Pull request by Devesh21700Kumar**
## Pull request checklist

Please check if your PR :rocket: fulfills the following requirements:
- [x] Build (`npm run start`) was run locally and new changes were working correctly
- [x] Lint (`npm run linter`) has passed locally and any fixes were made for failures

## Pull request type

 

Please check the type of change your PR introduces:
- [x] Feature
- [x] Code style update (formatting, renaming)

Fixes #

fixes #12 

## What is the new behavior?

-Adds discussion schemas and controllers
-Adds discussion services 
-CRUD operations mentioned below are possible

## Does this introduce a breaking change?

- [x] Yes

GET a new discussion by ID
image

ADD a new discussion
image

UPDATE a new discussion
image

DELETE a discussion
image

### CODE REVIEW

**Comment by Devesh21700Kumar** [comment]
@Sarvesh67 Pls review the final changes

File: src/modules/channelDiscussions/discussion.service.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+		await this.DiscussionRepo.update(DiscussionId, Discussion);
+		return await this.findById(DiscussionId);
+	}
+
+	@Transactional()
+	async deleteDiscussion(DiscussionId: number): Promise<Object | undefined> {
+		try{
+			const delRes = await this.DiscussionRepo.delete({ ID: DiscussionId });
+			if (delRes.affected===0) {
+				return{
```
Comment (Sarvesh67) [inline src/modules/channelDiscussions/discussion.service.ts]: Just throw not found exception (import from nest) here.

File: src/modules/channelDiscussions/discussion.service.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+			return Discussion;
+		}
+	}
+
+	@Transactional()
+	async updateDiscussion(
+		DiscussionId: number,
+		Discussion: UpdateDiscussionDto
+	): Promise<Discussions | undefined> {
+		await this.DiscussionRepo.update(DiscussionId, Discussion);
```
Comment (Sarvesh67) [inline src/modules/channelDiscussions/discussion.service.ts]: Handle the not found error here too. And remove findById statement. Just use the response of update method to throw exception like in deleteDiscussion method.

File: src/database/entity/channelDiscussion.ts [hunk 3]
```diff
@@ -10,7 +13,7 @@ import { Entity, PrimaryGeneratedColumn, Column, BaseEntity } from 'typeorm';
 @Entity({ name: 'discussions' })
 export class Discussions extends BaseEntity {
 	@PrimaryGeneratedColumn('increment', { type: 'int' })
-	message_ID?: Number;
+	ID?: Number;
```
Comment (Sarvesh67) [inline src/database/entity/channelDiscussion.ts]: Change 'ID' to 'id'. Keep it simple!

**Review by Sarvesh67 — CHANGES_REQUESTED** [review]

File: src/modules/channelDiscussions/discussion.service.ts (hunk 1 above)
Comment (Devesh21700Kumar) [reply src/modules/channelDiscussions/discussion.service.ts]: sure

File: src/database/entity/channelDiscussion.ts (hunk 3 above)
Comment (Devesh21700Kumar) [reply src/database/entity/channelDiscussion.ts]: sure

File: src/modules/channelDiscussions/discussion.service.ts (hunk 2 above)
Comment (Devesh21700Kumar) [reply src/modules/channelDiscussions/discussion.service.ts]: K

**Pull request merged**